src/main/java/bio/terra/service/dataset/AssetDao.java

```diff
@@ -178,7 +169,7 @@ private Collection<AssetTable> retrieveAssetTablesAndColumns(
           // add the new column to the asset table object
           assetTable.getColumns().add(newColumn);
         });
-    return tables.values();
+    return tables.values().stream().toList();
```

Comment:
I think `List.copyOf()` will work and should be more efficient
```suggestion
    return List.copyOf(tables.values());
```